### PR TITLE
initial changes to make it work

### PR DESCRIPTION
**Pull request by chihoyeung**

### CODE REVIEW

File: PowerDms.Api.Rest.Client.Demo/DemoConsoleApp/Program.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                    SiteKey = configuration["Credentials:SiteKey"],
+                    Username = configuration["Credentials:Username"],
+                    Password = configuration["Credentials:Password"],
+                    ClientSecret = configuration["Credentials:ClientSecret"]
+                };
+
+            var httpClient = PowerDmsHttpClientFactory.CreateHttpClient(configuration["Host"], ApiVersion.Version1);
+            var requestManager = new HttpRequestManagerFactory().CreateInstance(httpClient);
+
+            Console.WriteLine("Create Group - please delete from Site");
```
Comment (AlexDenton) [inline PowerDms.Api.Rest.Client.Demo/DemoConsoleApp/Program.cs]: This seems noisy as a part of the demo

File: PowerDms.Api.Rest.Client.Demo/DemoConsoleApp/Program.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                .AuthenticateWith(credentials);
+
+            var result = await requestManager.SendAsync(requestBuilder);
+
+            if (!result.IsSuccessStatusCode)
+            {
+                // CR note: why would I need to pass a <T> for errors?
+                var error = await result.GetErrorResponse<GroupDto>();
+                Console.WriteLine($"-- Error! Code: {error.Code}, Message: {error.Messages?.FirstOrDefault()} --");
+                return null;
```
Comment (AlexDenton) [inline PowerDms.Api.Rest.Client.Demo/DemoConsoleApp/Program.cs]: We should return `ServiceResponse`

File: PowerDms.Api.Rest.Client.Demo/DemoConsoleApp/Program.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+            var getBogusOne = await GetGroup(requestManager, credentials, "0");
+
+            Console.WriteLine("-- press any key to exit --");
+            Console.ReadKey();
+        }
+
+        private static async Task<GroupDto> GetGroup(HttpRequestManager requestManager, Credentials credentials, string groupId)
+        {
+            // I don't like the fact credentials need to be set on each builder, why not in the manager?
```
Comment (AlexDenton) [inline PowerDms.Api.Rest.Client.Demo/DemoConsoleApp/Program.cs]: I generally agree. I don't like that it leads you to create private methods for every call.

File: PowerDms.Api.Rest.Client/PowerDms.Api.Rest.Client/HttpAuthorization.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿namespace PowerDms.Api.Rest.Client
+{
+    /// <summary>
+    /// ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization
+    /// </summary>
+    public class HttpAuthorization
```
Comment (AlexDenton) [inline PowerDms.Api.Rest.Client/PowerDms.Api.Rest.Client/HttpAuthorization.cs]: Could maybe re-use AuthenticationHeaderValue

File: PowerDms.Api.Rest.Client/PowerDms.Api.Rest.Client/HttpResponseMessageExtensions.cs [hunk 5]
```diff
@@ -21,28 +24,54 @@ public static async Task<TResponse> AwaitGetSuccessfulResponse<TResponse>(this T
...
         {
-            if (!httpResponseMessage.IsSuccessStatusCode)
+            if (httpResponseMessage.IsSuccessStatusCode)
             {
                 throw new Exception();
             }
 
-            return await httpResponseMessage
-                .GetContent<TError>();
+            if (httpResponseMessage.Content.Headers.ContentType.MediaType == "application/json")
```
Comment (AlexDenton) [inline PowerDms.Api.Rest.Client/PowerDms.Api.Rest.Client/HttpResponseMessageExtensions.cs]: Don't like string. Might be some strongly typed dotnet header values.

File: PowerDms.Api.Rest.Client/PowerDms.Api.Rest.Client/RestApiRoutes.cs [hunk 6]
```diff
@@ -2,12 +2,13 @@
 {
     public class RestApiRoutes
     {
-        public const string PowerDmsRestApiDomain = "api.powerdms.com";
+        public const string PowerDmsRestApiDomain = "https://api.powerdms.com";
```
Comment (AlexDenton) [inline PowerDms.Api.Rest.Client/PowerDms.Api.Rest.Client/RestApiRoutes.cs]: We also need to make this extensible to work with non-SSL

File: PowerDms.Api.Rest.Client/PowerDms.Api.Rest.Client/HttpResponseMessageExtensions.cs (hunk 5 above)
Comment (chihoyeung) [reply PowerDms.Api.Rest.Client/PowerDms.Api.Rest.Client/HttpResponseMessageExtensions.cs]: one will be there in .net core 2.1

File: PowerDms.Api.Rest.Client/PowerDms.Api.Rest.Client/RestApiRoutes.cs (hunk 6 above)
Comment (chihoyeung) [reply PowerDms.Api.Rest.Client/PowerDms.Api.Rest.Client/RestApiRoutes.cs]: yes, I been testing it locally

**Pull request merged**